mlir-sycl/include/mlir/Dialect/SYCL/IR/SYCLOps.h

```diff
@@ -26,6 +26,37 @@ inline bool isSYCLOperation(Operation *op) {
   return isa<sycl::SYCLDialect>(op->getDialect());
 }
 
+/// Return true if type is 'memref<?x!sycl.accessor>'.
+inline bool isAccessorPtrType(Type type) {
+  auto mt = dyn_cast<MemRefType>(type);
+  bool isMemRefWithExpectedShape =
+      (mt && mt.hasRank() && (mt.getRank() == 1) &&
+       ShapedType::isDynamic(mt.getShape()[0]) && mt.getLayout().isIdentity());
+  if (!isMemRefWithExpectedShape)
+    return false;
+
+  return isa<sycl::AccessorType>(mt.getElementType());
+}
+
+/// Represent Value of type 'memref<?x!sycl.accessor>'.
+class AccessorPtr : public Value {
+public:
+  AccessorPtr(Value accessorPtr) : Value(accessorPtr) {
+    assert(classof(accessorPtr) &&
+           "Expecting to construct with an AccessorPtr");
```

Comment:
Suggest: "val should be an 'AccessorPtrValue'"

Reply:
done.